controllers/etcd_controller.go

```diff
@@ -889,12 +664,27 @@ func (r *EtcdReconciler) reconcileEtcd(ctx context.Context, logger logr.Logger,
 		return nil, nil, err
 	}
 
-	sts, err := r.reconcileStatefulSet(ctx, logger, etcd, values)
+	stsDeployer := componentsts.New(r.Client, etcd.Namespace, val.StatefulSet)
+	err = stsDeployer.Deploy(ctx)
+	// Ignore the precondition violated error, this machine is already updated
+	// with the desired label.
+	if err == errorsutil.ErrPreconditionViolated {
```

Comment:
I think you can remove this. You don't find it in the most of our components either and TBH I've never experienced it myself (it's not even used in the K8s org https://github.com/search?q=org%3Akubernetes+ErrPreconditionViolated&type=code).

Reply:
Removed